File: Proyecto/Escuela_circo/src/main/java/IFAEscuela_circo/Escuela_circo/Modelos/Curso.java
package IFAEscuela_circo.Escuela_circo.Modelos;

import jakarta.persistence.*;
import org.hibernate.annotations.ColumnDefault;

@Entity
@Table(name = "cursos")
public class Curso {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "codCurso", nullable = false, length = 5)
    private String codCurso;

    @Column(name = "nombre", nullable = false, length = 50)
    private String nombre;

    @Lob
    @Column(name = "descripcion")
    private String descripcion;

    @Lob
    @Column(name = "horario", nullable = false)
    private String horario;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "tipo", nullable = false)
    private Tipo tipo;

    @ColumnDefault("1")
    @Column(name = "activo", nullable = false)
    private Byte activo;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getCodCurso() {
        return codCurso;
    }

    public void setCodCurso(String codCurso) {
        this.codCurso = codCurso;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public String getHorario() {
        return horario;
    }

    public void setHorario(String horario) {
        this.horario = horario;
    }

    public Tipo getTipo() {
        return tipo;
    }

    public void setTipo(Tipo tipo) {
        this.tipo = tipo;
    }

    public Byte getActivo() {
        return activo;
    }

    public void setActivo(Byte activo) {
        this.activo = activo;
    }

}

File: Proyecto/AcademyGestor/AcademyGestor/Modelos/Curso.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AcademyGestor.Modelos
{
    class Curso
    {
        private int id;
        private string codCurso;
        private string nombre;
        private string descripcion;
        private string horario;
        private Tipo tipo;
        private bool activo;

        public int Id { get => id; set => id = value; }
        public string CodCurso { get => codCurso; set => codCurso = value; }
        public string Nombre { get => nombre; set => nombre = value; }
        public string Descripcion { get => descripcion; set => descripcion = value; }
        public string Horario { get => horario; set => horario = value; }
        public Tipo Tipo { get => tipo; set => tipo = value; }
        public bool Activo { get => activo; set => activo = value; }


    }
}
